src/core/Object3D.js

```diff
@@ -577,9 +646,13 @@ class Object3D extends EventDispatcher {
 
 	updateMatrix() {
 
-		this.matrix.compose( this.position, this.quaternion, this.scale );
+		if ( this.matrixNeedsUpdate === true ) {
 
-		this.matrixWorldNeedsUpdate = true;
+			this.matrix.compose( this.position, this.quaternion, this.scale );
+			this.matrixNeedsUpdate = false;
+			this.matrixWorldNeedsUpdate = true;
+
+		}
```

Comment:
Seems that moving `matrixWorldNeedsUpdate` inside of this guard breaks `CameraHelper` as seen in the `webgl_camera` example. Maybe should be `this.camera.matrixWorld.decompose( this.position, this.quaternion, this.scale );` instead inside of `CameraHelper.update`? Unsure if this is breaking, but the other properties are certainly read-only.

https://github.com/mrdoob/three.js/blob/27acd3126be340475227f2f4749d21e99f91b58b/src/helpers/CameraHelper.js#L110-L111

Reply:
Decomposing in 8c693277dbf88b2240df841d0592f2ddae0c976d looks fine. Not sure how widespread this pattern is though. If it's a problem that we assume Object3D members read-only, then we would have to concede optimizations to `this.matrixWorldNeedsUpdate = true;` here. Ideally, the world matrix is never updated if either the local matrix or parent world matrix didn't update.